### ISSUE TITLE
Set counter from absolute input

### ISSUE THREAD

**Issue by poggenpower** [fhemberger/mqtt_exporter#19]
By design `prometheus.Counter` only allows an increment and not setting it to a new absolute number. But in most cases you get not the increment via MQTT but a absolute number. see: https://www.robustperception.io/setting-a-prometheus-counter
Workaround could be to use `gauge` instead, but in this case the rollover or a reset of the count to `0` won't be recognizes correctly and will cause miss interpretation of the data e.g. when graphed in Grafana. 

My use case is a rain sensor that simple returns an ever up-counting number of ml/m2.

What I have read so far it requires an implementation of a Custom Collectors, not really found an easy solution for that as I am not a prometheus expert at all.

**Comment by poggenpower** [comment]
Here my code change https://github.com/poggenpower/mqtt_exporter/tree/issue19 to solve this. 
As it changes the logic a lot I am keen to get feedback.
Further I have not a real test case for "histogram" and "summary" please test if possible.

**Comment by fhemberger** [comment]
Looks interesting. Unfortunately, I don't have a matching setup to test your changes.
But if other users can give some feedback on your patch, I would consider merging it.

**Comment by poggenpower** [comment]
I have now created another fix for this issue with a new approach https://github.com/fhemberger/mqtt_exporter/pull/23. Even if I did some refactoring to make the code more readable and maintainable, they way how it works and calling the underlying prometheus functions stays the same. 
The new test-framework ensure the old behavior stays the same. 
Keen the get some feedback, other wise I will merge the code Monday or Tuesday.
